elastic/kibana
# [Ingest Pipelines] Add support for redact processor

Fixes: https://github.com/elastic/kibana/issues/150827

## Summary

Adds support for the [redact processor](https://www.elastic.co/guide/en/elasticsearch/reference/master/redact-processor.html).

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Ingest Pipelines` and create a new pipeline
3. Add new Redact processor that has:
    * field: `test`
    * params: `%{IP:REDACTED}` and `%{GITHUB_NAME:GITHUB_NAME}`
    * pattern definitions: `"GITHUB_NAME": "@%{USERNAME}"`
4. Verify that:
    * The request is correctly generated
    * Testing the pipeline with the following document should produce a valid redacted output
 ```
 [
  {
    "_source": {
      "test": "@pepe 55.3.244.1 GET /index.html 15824 0.043"
    }
  }
]
 ```
Should output:
```
{
  "docs": [
    {
      "doc": {
        "_index": "_index",
        "_id": "_id",
        "_version": "-3",
        "_source": {
          "test": "<GITHUB_NAME> <REDACTED> GET /index.html 15824 0.043"
        },
        "_ingest": {
          "timestamp": "2023-02-28T14:47:53.89462Z"
        }
      }
    }
  ]
}
```

<details>
<summary>Screenshots</summary>

![Screenshot 2023-03-01 at 13 11 50](https://user-images.githubusercontent.com/1191206/222136041-452b9bd3-5839-415c-a9ac-0b432efa6ad4.png)
![Screenshot 2023-03-01 at 13 11 59](https://user-images.githubusercontent.com/1191206/222136054-ab7f5b80-a1f1-4b50-a7cd-fc52bb2d1063.png)


![Screenshot 2023-02-28 at 15 10 11](https://user-images.githubusercontent.com/1191206/221887912-094f946c-dea1-48e7-b6a0-675d24f9b8f4.png)
![Screenshot 2023-02-28 at 15 10 48](https://user-images.githubusercontent.com/1191206/221887905-df72c6bb-ac89-45ca-989d-5ead6c838eed.png)

</details>
